sql/select.go

```diff
@@ -124,11 +126,16 @@ func (p *planner) selectIndex(s *scanNode, ordering []int) (planNode, error) {
 	}
 
 	if s.filter != nil {
+		if group != nil {
+			// Allow the group-by to add an implicit "IS NOT NULL" filter.
+			s.filter = group.isNotNullFilter(s.filter)
+		}
+
 		// Analyze the filter expression, simplifying it and splitting it up into
 		// possibly overlapping ranges.
-		exprs := analyzeExpr(s.filter)
+		exprs, equivalent := analyzeExpr(s.filter)
 		if log.V(2) {
-			log.Infof("analyzeExpr: %s -> %s", s.filter, exprs)
+			log.Infof("analyzeExpr: %s -> %s [equivalent=%v]", s.filter, exprs, equivalent)
```

Comment:
`%v` also prints `{true,false}` for boolean values and I've used it frequently for printing booleans in the code base. Do you think the explicitness of `%t` is necessary?